app/components/Navbar/MobileNavbar.tsx

```diff
@@ -1,43 +1,54 @@
 'use client'
 import React, { useEffect, useState } from 'react'
-import { NAVBAR_ITEMS, Item } from './fixture'
-import Image from 'next/image'
+import { NAVBAR_ITEMS, Item, NavbarItem } from './fixture'
 import Link from 'next/link'
 import { X, Discord } from '../icons'
 import { Drawer, CustomFlowbiteTheme } from 'flowbite-react'
 import { isExternalLink } from '.'
-import { EtherlinkLogo } from '../EtherlinkLogo'
+import { CheveronIcon } from '../Icons/CheveronIcon'
+import { ArrowRightIcon } from '../Icons/ArrowRightIcon'
+import { PrimaryButton } from '../buttons/PrimaryButton'
 
 const customDrawerTheme: CustomFlowbiteTheme['drawer'] = {
   root: {
-    base: 'fixed z-40 overflow-y-auto bg-white p-4 transition-transform duration-300 pt-8 bg-darkBlack',
-    backdrop: '',
-    edge: 'bottom-16',
+    base: 'fixed z-40 overflow-y-auto bg-white p-4 transition-transform duration-300 pt-0 bg-grey900 rounded-t-3xl border-t border-grey600',
+    backdrop: 'fixed inset-0 z-30 bg-gray-900/50 bg-transparent',
+    edge: 'bottom-0',
     position: {
       right: {
-        on: 'right-0 top-0 h-screen w-full transform-none',
-        off: 'right-0 top-0 h-screen w-80 translate-x-full'
+        on: 'right-0 bottom-0 h-auto w-full transform-none',
+        off: 'right-0 bottom-0 h-auto w-full translate-y-full'
       }
     }
   },
   header: {
     inner: {
       closeButton:
-        'absolute end-6 top-9 flex h-8 w-8 items-center justify-center rounded-lg bg-transparent text-sm text-white hover:bg-darkBlack',
+        'absolute end-10 top-8 flex h-6 w-4 items-center justify-center rounded-lg bg-transparent text-grey100 text-sm hover:bg-darkBlack',
       closeIcon: 'h-6 w-6'
     }
   },
   items: {
     base: ''
   }
 }
+
 export const MobileNavbar = ({
   isOpen,
   handleClose
 }: {
   isOpen: boolean
   handleClose: () => void
 }) => {
+  const [currentView, setCurrentView] = useState<'main' | 'dropdown'>('main')
+  const [activeDropdown, setActiveDropdown] = useState<Item[] | null>(null)
+  const [openDropdowns, setOpenDropdowns] = useState<{
+    [key: number]: boolean
+  }>({})
+  const [selectedItemTitle, setSelectedItemTitle] = useState<string>('')
+  const [isAnimating, setIsAnimating] = useState(false)
```

Comment:
why do we need isAnimating? it seems working fine without this state

Reply:
Don't want to mess up with the animation if clicked quickly back and forth, let's leave it.